### PR TITLE
ISPN-6471 Avoid blocking Netty IO threads, delegating to a separate executor

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-6471

PREVIEW ONLY

Working on bechmark tests

### CODE REVIEW

File: server/core/src/main/scala/org/infinispan/server/core/transport/NettyChannelInitializer.scala [hunk 1]
```diff
@@ -16,17 +18,17 @@ import io.netty.handler.ssl.SslHandler
...
+                              encoder: ChannelOutboundHandler, executor: EventExecutorGroup)
+        extends ChannelInitializer[Channel] {
 
    override def initChannel(ch: Channel): Unit = {
       val pipeline = ch.pipeline
       val ssl = server.getConfiguration.ssl
       if (ssl.enabled())
-         pipeline.addLast("ssl", new SslHandler(createSslEngine(ssl)))
-      pipeline.addLast("decoder", server.getDecoder)
+         pipeline.addLast(executor, "ssl", new SslHandler(createSslEngine(ssl)))
```
Comment (wburns) [inline server/core/src/main/scala/org/infinispan/server/core/transport/NettyChannelInitializer.scala]: Do we want this for ssl as well?

**Comment by wburns** [comment]
I was also planning on doing this with my changes.  It would be nice to see what benchmarks you are using, as I will need them for other purposes as well :)

File: server/core/src/main/scala/org/infinispan/server/core/transport/NettyChannelInitializer.scala (hunk 1 above)
Comment (gustavonalle) [inline server/core/src/main/scala/org/infinispan/server/core/transport/NettyChannelInitializer.scala]: Yes, any concerns?

**Comment by gustavonalle** [comment]
There directory memory failures in the ```hotrod-client``` suite, will investigate

@wburns The idea is to use a JMH benchmark hammering 1 server, doing a mix of fast ops (get/put) and potentially blocking ops (script, file system based indexing)

**Comment by gustavonalle** [comment]
Closing since changes are incorporated in https://github.com/infinispan/infinispan/pull/4240

**Pull request closed without merging**